### PR TITLE
[MRG] FIX: allow reading experimental dipole

### LINKED ISSUE #420: read_dipole fails with file in HNN repository
Is the `read_dipole` function supposed to work with this file:

https://github.com/jonescompneurolab/hnn/blob/master/data/MEG_detection_data/S1_SupraT.txt 

?

When I do:

```py
from urllib.request import urlretrieve
from hnn_core.dipole import read_dipole

data_url = ('https://raw.githubusercontent.com/jonescompneurolab/hnn/master/'
            'data/MEG_detection_data/S1_SupraT.txt')
urlretrieve(data_url, 'S1_SupraT.txt')
exp_dpl = read_dipole('S1_SupraT.txt')
```

it throws me an error. Thoughts @rythorpe @ntolley ?

### PR DESCRIPTION
**Pull request by jasmainak**
closes #420

### CODE REVIEW

File: hnn_core/dipole.py [hunk 1]
```diff
@@ -256,7 +256,15 @@ class Dipole(object):
 
     def __init__(self, times, data, nave=1):  # noqa: D102
         self.times = np.array(times)
-        self.data = {'agg': data[:, 0], 'L2': data[:, 1], 'L5': data[:, 2]}
+
+        if data.ndim == 1:
+            data = data[:, None]
```
Comment (rythorpe) [inline hnn_core/dipole.py:261]: What's this for?

File: hnn_core/dipole.py [hunk 2]
```diff
@@ -137,15 +137,14 @@ def average_dipoles(dpls):
...
-
-    avg_dpl = Dipole(dpls[0].times, avg_dpl_data)
+    avg_data = list()
+    layers = dpl.data.keys()
+    for layer in layers:
+        avg_data.append(
+            np.mean(np.array([dpl.data[layer] for dpl in dpls]), axis=0)
+        )
+    avg_data = np.c_[avg_data].T
+    avg_dpl = Dipole(dpls[0].times, avg_data)
```
Comment (rythorpe) [inline hnn_core/dipole.py:147]: This makes me wonder why weren't doing this all along. Looks great!

File: hnn_core/dipole.py (hunk 1 above)
Comment (jasmainak) [reply hnn_core/dipole.py:261]: If you provide a 1D array, then data.shape[1] doesn't exist down below. For e.g., data.shape = (6000,) will become (6000, 1)

Comment (jasmainak) [reply hnn_core/dipole.py:261]: I tried using `np.atleast_2d` first but then it made it (1, 6000) so I just went with my manual solution

Comment (rythorpe) [reply hnn_core/dipole.py:261]: I see, that makes sense!

**Review by rythorpe — APPROVED** [review]
Does anyone else want to take a look or shall we merge?

**Comment by rythorpe** [comment]
I'm going to go ahead and merge this to maintain forward momentum. Thanks @jasmainak!

**Pull request merged**